default_gateway.go

```diff
@@ -103,6 +103,9 @@ func (sb *sandbox) needDefaultGW() bool {
 		if ep.getNetwork().Type() == "null" || ep.getNetwork().Type() == "host" {
 			continue
 		}
+		if ep.joinInfo.disableGatewayService {
+			return false
+		}
```

Comment:
> When a driver sets disableGatewayService, i assume that the driver is capable of providing the default route for that container.

Not necessarily. Earlier this assumption was ok because the check was specifically for Gateway IP. Now a driver can set disableGatewayService and doesn't provide gateway IP or static default route.

I am not against this change. But want to make sure there is agreement on the behavior with this change.